packages/autorest.typescript/src/restLevelClient/transforms/transformResponseTypes.ts

```diff
@@ -0,0 +1,144 @@
+            // We'll need http information such as headers, statusCodes, etc continue if not available
+            if (!response.protocol.http) {
+                continue;
+            }
+            let schemaResponse: SchemaResponse = responseToSchemaResponse(response);
+            const rlcResponseUnit: ResponseMetadata = {
+                statusCode: getStatusCode(schemaResponse),
+                description: operationLanguageMetadata.description
+            };
+
+            // transform header
+            transformHeaders(response, rlcResponseUnit);
+            // transform body
+            transformBody(response, importedModels, rlcResponseUnit);
+            rlcOperationUnit.responses.push(rlcResponseUnit);
+        }
+        rlcResponses.push(rlcOperationUnit);
+    }
+    if (importedModels.size > 0) {
+        importDetails.set(ImportKind.ResponseOutput, importedModels);
+    }
+    return rlcResponses;
+}
+
+
+function transformHeaders(from: Response, to: ResponseMetadata) {
+    // Check if there are any required headers
+    const hasDefinedHeaders =
+        Boolean(from.protocol.http?.headers) &&
+        Boolean(from.protocol.http?.headers.length);
+    if (!hasDefinedHeaders) {
+        return;
+    }
+
+    to.headers = from.protocol.http?.headers.map((h: HttpHeader) => {
+        const header: HeaderMetadata = {
+            name: `"${h.header.toLowerCase()}"`,
+            description: getLanguageMetadata(h.language).description,
+            type: primitiveSchemaToType(h.schema, [
+                SchemaContext.Output,
+                SchemaContext.Exception
+            ]),
+            required: false
+        };
+        return header;
+    });
+}
+
+function transformBody(from: Response, importedModels: Set<string>, to: ResponseMetadata) {
+    let schemaResponse: SchemaResponse = responseToSchemaResponse(from);
+    const bodyType = getBodyTypeName(schemaResponse, importedModels);
+    const bodyDescription = getFormatDocs(schemaResponse.schema);
+    if (!bodyType) {
+        return;
+    }
+    to.body = {
+        name: "body",
+        type: bodyType,
+        description: bodyDescription
+    };
+}
+
+/**
+ * Body types are defined in the models file, this function checks if the current
+ * response's body has a reference to a model or if it is a primitive, and returns the Typescript type
+ * to generate
+ * @param response - response to get the body type from
+ * @param importedModels - track models to import
+ */
+function getBodyTypeName(
+    response: SchemaResponse,
+    importedModels: Set<string>
+): string | undefined {
+    return getElementType(
+        response.schema,
+        [SchemaContext.Output, SchemaContext.Exception],
+        importedModels
+    );
+}
+
+// Gets a list of all the available operations requests in the specification
+function getAllOperationRequests(model: CodeModel) {
+    let operations: Operation[] = [];
+    model.operationGroups.forEach(og =>
```

Comment:
nit: maybe this would be more readable

```ts
for (const og of model.operationGroups) {
   for (const operation of og.operations) {
      if (operation.requests?.length) {
        operations.push(operation); 
     }
   }
}
```